empirical_photometry.ipynb

```diff
@@ -0,0 +1,448 @@
+    "    bp = SpectralElement(Empirical1D,\n",
+    "                         points=filt.wavelength,\n",
+    "                         lookup_table=filt.transmittance)\n",
+    "    # for boxy bandpasses, tapering them with taper() is recommended\n",
+    "    bandpasses[band] = bp.taper()\n",
+    "\n",
+    "    plt.plot(bp.waveset, bp(bp.waveset), label=band)\n",
+    "plt.legend(loc='upper right')\n",
+    "plt.show()\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can also construct a bandpass from a file using <tt>synphot</tt>'s <a href=\"https://synphot.readthedocs.io/en/latest/api/synphot.spectrum.SpectralElement.html#synphot.spectrum.SpectralElement.from_file\">SpectralElement.from_file()</a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Once we have all of the effects on the source spectrum set, we can combine them together by simply multiplying the SpectralElement objects:"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Model the observation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from synphot.observation import Observation\n",
+    "\n",
+    "# zero point flux of sdss system\n",
+    "zero_point_star_equiv = u.zero_point_flux(3631.1 * u.Jy)\n",
+    "\n",
+    "fluxs = {}\n",
+    "for band in sdss_bands:\n",
+    "    observation = Observation(spectrum, bandpasses[band], force='taper')\n",
+    "    flux = observation.effstim('Jy')\n",
+    "    fluxs[band] = flux.to(u.nanomaggy, zero_point_star_equiv)\n",
+    "\n",
+    "# # SDSS mags aren't exactly equal to AB mags. Apply corrections\n",
+    "# mags['u'] += 0.04\n",
+    "# mags['z'] -= 0.02"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "u = 8.98588 nmgy\n",
+      "g = 144.40355 nmgy\n",
+      "r = 256.01483 nmgy\n",
+      "i = 397.2675 nmgy\n",
+      "z = 362.74829 nmgy\n"
+     ]
+    }
+   ],
+   "source": [
+    "for band in sdss_bands:\n",
+    "    print(band + ' =', np.round(fluxs[band], 5))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Compare the u,g,r,i,z empirical magnitudes to what we predict with synphot"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Get the magnitudes measured by the SDSS fibers"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 45,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model_flux = ['fiberFlux_' + band for band in sdss_bands]\n",
+    "flux_table = SDSS.query_crossid(coordinates = coords, photoobj_fields=model_flux)\n",
+    "sdss_fluxs = {}\n",
+    "for band in sdss_bands:\n",
+    "    # sdss fluxes are given in units of \"nanomaggies\"\n",
```

Comment:
Perhaps you can leave an inline comment here with [this link](http://www.sdss3.org/dr8/algorithms/magnitudes.php), it's a pretty good narrative explanation of what the heck this is all about.